packages/portal-proto/src/features/cases/FilesTable.tsx

```diff
@@ -0,0 +1,380 @@
+import { FileAccessBadge } from "@/components/FileAccessBadge";
+import FunctionButton from "@/components/FunctionButton";
+import VerticalTable from "@/components/Table/VerticalTable";
+import { HandleChangeInput } from "@/components/Table/types";
+import { TableActionButtons } from "@/components/TableActionButtons";
+import { HeaderTitle } from "@/components/tailwindComponents";
+import {
+  capitalize,
+  fileInCart,
+  statusBooleansToDataStatus,
+} from "@/utils/index";
+import {
+  AccessType,
+  GdcFile,
+  SortBy,
+  selectCart,
+  useCoreDispatch,
+  useCoreSelector,
+  useGetFilesQuery,
+} from "@gff/core";
+import { Tooltip } from "@mantine/core";
+import {
+  ColumnDef,
+  ColumnOrderState,
+  SortingState,
+  VisibilityState,
+  createColumnHelper,
+} from "@tanstack/react-table";
+import fileSize from "filesize";
+import Link from "next/link";
+import { useEffect, useState } from "react";
+import { useDeepCompareEffect, useDeepCompareMemo } from "use-deep-compare";
+import { mapGdcFileToCartFile } from "../files/utils";
+import download from "@/utils/download";
+import { downloadTSV } from "@/components/Table/utils";
+import { convertDateToString } from "@/utils/date";
+
+interface FilesTableProps {
+  caseId: string;
+}
+
+type CaseFilesTableDataType = {
```

Comment:
Could you define this as a pick type of GdcFile like this? https://github.com/NCI-GDC/gdc-frontend-framework/blob/ca5032df7da7d503631d1ee7b60c52175fef5bd2/packages/core/src/features/cart/updateCartSlice.ts#L6

Reply:
This is changed.